bin/check-s3-bucket-visibility.rb

```diff
@@ -60,6 +74,26 @@ def s3_client
     @s3_client ||= Aws::S3::Client.new
   end
 
+  def s3_resource
+    @s3_resource || Aws::S3::Resource.new
+  end
+
+  def list_buckets
+    buckets = []
+    s3_resource.buckets.each do |bucket|
+      if s3_resource.client.get_bucket_location(bucket: bucket.name).location_constraint == config[:aws_region]
```

Comment:
if `get_bucket_location` fails on the bucket, should we catch/handle that? (I assume that's possible based on the bucket policy setup)

Reply:
I did not see much in terms of issues other than trying to get a location for a bucket in another region which I solved by ensuring that the region matched. I grabbed it from: https://docs.aws.amazon.com/sdk-for-ruby/v3/developer-guide/s3-example-get-buckets-in-region.html#aws-ruby-sdk-s3-example-get-buckets-in-region I can't think of what other scenerios there might be and aws sdk makes finding all their error classes a PITB because of all the layers of abstraction. We certainly won't have the common issues like bucket not found as this only gets called when `--all-buckets true` is passed and we get what the aws api returns unlike if you are passing a bucket in manually via `--bucket-names` as you could easily typo that. If you can find some specific error classes to rescue on I am more than happy to add them.

Reply:
Off the cuff, I can't remember, I feel like we hit something like this internally. That said, I'm 100% game to get this in there, and I can throw it at our weird policy buckets and PR updates if I find anything.

Reply:
the error I remember having was grabbing the policies for a bucket outside of the region specified. Rather than rescue it there I decided to just make this function return data that makes sense given the context of this check.